### PR TITLE
added add playlist add video to playlist

### PR DESCRIPTION
**Pull request by Hnh013**
added Playlist page and service methods for playlist
added add a playlist
added add video to a playlist

### CODE REVIEW

**Comment by Hnh013** [comment]
added delete a playlist
added remove video from a playlist

File: src/frontend/contexts/dataContext.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const DataContext = createContext();
 
 const DataProvider = ({ children }) => {
     let [ videos , setVideos] = useState([]);
+    let [ selectedVideo , setSelectedVideo ] = useState('none');
```
Comment (OmKakatkar) [inline src/frontend/contexts/dataContext.js]: You can use `const` here as well, `let` doen't seem to be required here

File: src/frontend/contexts/userContext.js [hunk 2]
```diff
@@ -26,6 +26,14 @@ const UserProvider = ({children}) => {
...
             case 'REMOVEFROMWATCHLATER': 
                 return { ...state, foundUser : action.payload };
+            case 'ADDPLAYLIST': 
+                return { ...state, foundUser : action.payload };
+            case 'REMOVEPLAYLIST': 
+                return { ...state, foundUser : action.payload };
+            case 'ADDVIDEOTOPLAYLIST': 
+                return { ...state, foundUser : action.payload };
+            case 'REMOVEVIDEOFROMPLAYLIST': 
+                return { ...state, foundUser : action.payload };
```
Comment (OmKakatkar) [inline src/frontend/contexts/userContext.js]: Seems like the functionality is being repeated everywhere. You can combine it all together. Like below:

```case 'REMOVEPLAYLIST':   
 case 'ADDVIDEOTOPLAYLIST':   
 case 'REMOVEVIDEOFROMPLAYLIST':   
     return { ...state, foundUser : action.payload };
```

**Review by OmKakatkar — APPROVED** [review]
Functionality is working as expected.
Variable names are consistent

Suggestions: Hide the floating card of playlist management when clicked outside of it.

File: src/frontend/templates/pages/Playlists.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { useUser } from '../../contexts/userContext';
+import { removePlaylist } from '../../services/playlistService';
+import { toastError, toastSuccess } from '../../services/toastService';
+import { Video } from '../components';
+
+export const Playlists = () => {
+
+    const { userState , userDispatcher } = useUser();
+
+    const RP = async (myToken, playlistDetails) => {
```
Comment (OmKakatkar) [inline src/frontend/templates/pages/Playlists.js]: Be more descriptive in for this variable. `RP` doesn't seem to be meaningful. `reactPlayer` can be used here

File: src/frontend/contexts/dataContext.js (hunk 1 above)
Comment (Hnh013) [reply src/frontend/contexts/dataContext.js]: hi Om, surely will look into it and change it

File: src/frontend/contexts/userContext.js (hunk 2 above)
Comment (Hnh013) [reply src/frontend/contexts/userContext.js]: yes Om , yes would do this this asap

File: src/frontend/templates/pages/Playlists.js (hunk 3 above)
Comment (Hnh013) [reply src/frontend/templates/pages/Playlists.js]: Hi om, thanks for this ,  i have made changes and given meaningful names to functions

File: src/frontend/contexts/userContext.js [hunk 4]
```diff
@@ -8,23 +8,21 @@ const UserProvider = ({children}) => {
 
     const userStateUpdater = (state, action) => {
         switch(action.type) {
-            case 'LOGIN': 
+            case 'LOG_IN':
```
Comment (mdarif6) [inline src/frontend/contexts/userContext.js:11]: - I Liked user-context structure

**Review by mdarif6 — APPROVED** [review]
- multiple place console.log is there- you can remove if not required.
- folder structure is easy to read

**Comment by Hnh013** [comment]
> * multiple place console.log is there- you can remove if not required.
> 
>     * folder structure is easy to read

hi Arif , thanks for the review , console.log I have used in catch error during signup and login just in case an error arises  , on one other place it was left by mistake ill remove it .

**Review by Aman-Jain1996 — COMMENTED** [review]
everything seems good , You can club the changes under 1 case in reducer as suggested by Om

**Pull request merged**